mas-foundation/src/main/java/com/ca/mas/foundation/MASDeviceMetadata.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (c) 2016 CA. All rights reserved.
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license.  See the LICENSE file for details.
+ *
+ */
+package com.ca.mas.foundation;
+import com.ca.mas.core.MobileSsoConfig;
+import com.ca.mas.core.store.StorageProvider;
+import com.ca.mas.foundation.notify.Callback;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+public class MASDeviceMetadata {
```

Comment:
This is more an internal or utility class, if possible make it package access and rename the class without MAS as the prefix, the MAS prefix is more for developer facing interface.